src/m365/purview/commands/retentioneventtype/retentioneventtype-remove.ts

```diff
@@ -0,0 +1,107 @@
+import auth from '../../../../Auth';
+import { Cli } from '../../../../cli/Cli';
+import { Logger } from '../../../../cli/Logger';
+import GlobalOptions from '../../../../GlobalOptions';
+import { validation } from '../../../../utils/validation';
+import request, { CliRequestOptions } from '../../../../request';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+import { accessToken } from '../../../../utils/accessToken';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  id: string;
+  confirm?: boolean;
+}
+
+class PurviewRetentionEventTypeRemoveCommand extends GraphCommand {
+  public get name(): string {
+    return commands.RETENTIONEVENTTYPE_REMOVE;
+  }
+
+  public get description(): string {
+    return 'Delete a retention event type';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        confirm: !!args.options.confirm
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-i, --id <id>'
+      },
+      {
+        option: '--confirm'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (!validation.isValidGuid(args.options.id)) {
+          return `'${args.options.id}' is not a valid GUID.`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (accessToken.isAppOnlyAccessToken(auth.service.accessTokens[this.resource].accessToken)) {
+      this.handleError('This command does not support application permissions.');
```

Comment:
Should we add a return statement so the runtime stops evaluating the rest of the code?

Reply:
This should be done automatically by the `handleError` method:
![image](https://user-images.githubusercontent.com/9193870/216951927-055a05cc-2db1-464c-aed1-8df3268040e3.png)